src/Indigo/Contracts/HomebaseLite/Impl/Admin.hs

```diff
@@ -15,53 +16,86 @@ import Indigo hiding ((*))
 
 import Morley.Util.Interpolate (itu)
 
-setAdmin :: tp :~> Address => IndigoEntrypoint tp
-setAdmin _ = do
+import Indigo.Contracts.HomebaseLite.Types
+import Indigo.Contracts.HomebaseLite.Impl.Utils
+
+checkSenderIsAdmin :: HasStorage Storage => IndigoM ()
```

Comment:
:thinking: I just realized a disadvantage of having a single flat/`plain` `Parameter` type: this needs to be called multiple times.

If we had a sub-parameter with all the entrypoints that required the `admin` role we could make this check once (before moving on to matching on the actual entrypoints). 

Quite the annoying penalty, we should probably reconsider our previous choice.
However, given that it might be best to update the dependencies for this to work well with autodoc, perhaps it's worth leaving this for a future optimization issue/PR.